### PR TITLE
Endpoints

### PR DESCRIPTION
**Pull request by samlandfried**

### CODE REVIEW

File: test/endpoints/get-test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  before(done => {
+    this.port = 9876;
+    this.server = app.listen(this.port, (error, result) => {
+      if (error) { done(error); }
+
+      done();
+    });
+  });
+
+  beforeEach(done => {
```
Comment (samlandfried) [inline test/endpoints/get-test.js]: @tmikeschu This feels so wrong (Nesting each DB insertion in a separate callback)... Is this what the async library you were talking about handles?

Comment (tmikeschu) [inline test/endpoints/get-test.js]: @samlandfried you're lucky: stuck at the airport! `async` and `await` are baked into ES7 (I believe), and are already compatible with most browsers.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
https://medium.com/@tmvvr/ecmascript-async-await-to-the-rescue-fc379ff89146

File: lib/models/food.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+const destroy = name => {
+  return db.raw(
+    'DELETE FROM foods WHERE name = ?', [name]
+  );
+};
+
+const update = (id, params) => {
+  const name = params.name;
+  const calories = params.calories;
+
```
Comment (samlandfried) [inline lib/models/food.js]: @tmikeschu Is there a better SQL way to do this?

**Comment by samlandfried** [comment]
@tmikeschu This PR completes the backend.

File: lib/models/food.js (hunk 2 above)
Comment (tmikeschu) [inline lib/models/food.js]: @samlandfried you could do something like:

```js
Object.keys(params).forEach(param => {
  param && db.raw("UPDATE foods SET `param` = ? WHERE id= ?", [params[param], id]
})

return db.raw("SELECT * FROM foods WHERE id = ?", [id])
```

**Pull request merged**